### PR TITLE
Add income amount adjustment deduction (所得金額調整控除)

### PR DESCRIPTION
**Pull request by shakuzen**
Fixes #344.

## What

Applies the **所得金額調整控除（子ども・特別障害者等を有する者等）** (income amount adjustment deduction) for taxpayers with employment income over **¥8,500,000** who have a qualifying dependent. Up to **¥150,000** — `⌈{min(salary, ¥10M) − ¥8.5M} × 10%⌉` — is subtracted from 給与所得.

Previously, net employment income and taxable income were overstated by up to ¥150,000 for these households, inflating both income tax and residence tax and, because 合計所得金額 feeds several eligibility thresholds, wrongly denying things like the home loan tax credit at its ¥20M income limit. (Pre-existing; surfaced while cross-checking #307.)

## Why fold it into net employment income (and not show it as a separate 所得控除)

This is technically correct, not just convenient. 措法41の3の11 defines 給与所得の金額 itself as `(収入 − 給与所得控除) − 所得金額調整控除`, so it reduces 給与所得 by statute — it is **not** in the 所得控除 bucket (基礎控除/扶養控除/etc.). This matches:
- the 源泉徴収票 box **給与所得控除後の金額（調整控除後）**, which is already net of it; and
- the 確定申告書, where it is folded into the 給与 所得金額 line, not the deductions section.

Applying it once, right after the 給与所得控除, means the reduced 合計所得金額 flows consistently into every downstream consumer: income-tax & residence-tax taxable income, both basic-deduction tier lookups, the residence-tax 調整控除 ¥25M cutoff, the home-loan-credit income limit, spouse/dependent thresholds, the NHI base, and the furusato limit. It creates no 人的控除差, so the residence-tax 調整控除 statutory-difference logic is intentionally untouched.

**Applies to both income tax and residence tax**, because it lowers the shared 合計所得金額 base.

## Eligibility

Derived from existing dependent data:
- **ロ** a 扶養親族 under 23 (includes under-16, unlike 扶養控除) within the income threshold
- **ハ** a special-disability 同一生計配偶者 or 扶養親族 within the income threshold

Not modeled (documented in tooltip + CHANGELOG): **イ** the taxpayer's own special-disability status (needs a new input field), and the 給与・年金 both-income variant (措法41の3の12).

## UI

The Net Employment Income tooltip (Taxes tab and Social Insurance tab) now shows the adjustment as its own line, plus an explanatory note with the NTA source — only when it applies. The final "Net Employment Income" figure equals the 源泉徴収票's 調整控除後 box; the tooltip just breaks out the two components.

## Testing

- 534 tests pass (new unit + integration tests cover the formula/eligibility edge cases, the worked example, and the home-loan-credit flip at the ¥20M limit); `tsc -b` and ESLint clean.
- Browser-verified the worked example (¥22M salary + child under 23): Net Employment Income ¥20,050,000 → **¥19,900,000**, subtotal taxable income down exactly ¥150,000, both income tax and residence tax decrease, tooltip reconciles. No adjustment row when there's no qualifying dependent.

### CODE REVIEW

File: src/components/TakeHomeCalculator/InputForm.tsx [hunk 1]
```diff
@@ -447,8 +447,9 @@ export const TakeHomeInputForm: React.FC<TaxInputFormProps> = ({ inputs, onInput
     }
   };
 
-  // We only need the total net income for the dependents modal.
-  const taxpayerNetIncome = React.useMemo(() => calculateTotalNetIncome(inputs.incomeStreams), [inputs.incomeStreams]);
+  // We only need the total net income for the dependents modal. Pass dependents so the
+  // 所得金額調整控除 is reflected, keeping the modal's eligibility hints consistent with the results.
+  const taxpayerNetIncome = React.useMemo(() => calculateTotalNetIncome(inputs.incomeStreams, undefined, inputs.dependents), [inputs.incomeStreams, inputs.dependents]);
```
Comment (shakuzen) [inline src/components/TakeHomeCalculator/InputForm.tsx:452]: Passing `undefined` for the year is fine for now. This will be addressed in a follow-up as there are multiple places this handling needs to be fixed.
